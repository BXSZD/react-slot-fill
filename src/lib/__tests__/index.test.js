import React from 'react';
import renderer from 'react-test-renderer';
import { Fill, Slot, Provider } from '../';

const Toolbar = (props) =>
  <div>
    <Slot name="Toolbar.Item" />
  </div>

Toolbar.Item = ({ label }) =>
  <Fill name="Toolbar.Item">
    <button>{label}</button>
  </Fill>

const Footer = (props) =>
  <div>
    <Slot name="Footer.Item" />
  </div>

Footer.Item = ({ href, label }) =>
  <Fill name="Footer.Item">
    <a href={href}>{label}</a>
  </Fill>

it('Fills the a simple slot', () => {
  const Feature = () => <Toolbar.Item label="Home 1" />;

  const fillComponent = renderer.create(
    <Provider>
      <Toolbar />
      <Feature />
    </Provider>
  );

  expect(fillComponent).toMatchSnapshot();
})

it('Fills the appropriate slot', () => {
  const Feature = () => [
    <Toolbar.Item label="Home 2" />,
    <Toolbar.Item label="About" />,
    <Footer.Item label="Twitter" href="twitter.com/reactjs" />,
  ];

  const fillComponent = renderer.create(
    <Provider>
      <Toolbar />
      <Footer />
      <Feature />
    </Provider>
  );

  expect(fillComponent).toMatchSnapshot();
});