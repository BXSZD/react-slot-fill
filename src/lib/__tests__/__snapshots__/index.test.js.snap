exports[`test Fills the a simple slot 1`] = `
<div>
  <button>
    Home 1
  </button>
</div>
`;

exports[`test Fills the appropriate slot 1`] = `
Array [
  <div>
    <button>
        Home 2
    </button>
    <button>
        About
    </button>
</div>,
  <div>
    <a
        href="twitter.com/reactjs">
        Twitter
    </a>
</div>,
]
`;
